src/components/app/details/appDetails/AppStatusModal.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from 'react';
+import { ReactComponent as Close } from '../../../../assets/icons/ic-close.svg';
+import { AppStreamData, AggregatedNodes } from '../../types';
+import { Drawer } from '../../../common';
+
+export const AppStatusModal: React.FC<{
+    streamData: AppStreamData;
+    nodes: AggregatedNodes;
+    appName: string;
+    environmentName: string;
+    status: any;
+    close: (...args) => void;
+    message: string;
+}> = ({ streamData, nodes, status, close, message, appName, environmentName }) => {
+    const [nodeStatusMap, setNodeStatusMap] = useState(new Map());
+    useEffect(() => {
+        const stats = streamData?.result?.application?.status?.operationState?.syncResult?.resources?.reduce(
+            (agg, curr) => {
+                agg.set(`${curr.kind}/${curr.name}`, curr);
+                return agg;
+            },
+            new Map(),
+        );
+        setNodeStatusMap(stats);
+    }, [streamData]);
+
+    function getNodeMessage(kind, name) {
+        if (nodeStatusMap && nodeStatusMap.has(`${kind}/${name}`)) {
+            const { status, message } = nodeStatusMap.get(`${kind}/${name}`);
+            if (status === 'SyncFailed') return 'Unable to apply changes: ' + message;
```

Comment:
we are showing message only if its SyncFailed otherwise we dont show any message?

Reply:
yes. I have not changed the logic.

Reply:
ok